src/find-best-model/src/main/scala/EvaluationUtils.scala

```diff
@@ -0,0 +1,119 @@
+      case cls: TrainedClassifierModel => SchemaConstants.ClassificationKind
+      case evm: BestModel => getTransformerModelType(evm.getBestModel)
+      case _ => throw new Exception(modelTypeUnsupportedErr)
+    }
+  }
+
+  def getEstimatorModelType(model: Estimator[_]): String = {
+    model match {
+      case reg: TrainRegressor => SchemaConstants.RegressionKind
+      case cls: TrainClassifier => SchemaConstants.ClassificationKind
+      case _ => throw new Exception(modelTypeUnsupportedErr)
+    }
+  }
+
+  def getMetricWithOperator(model: Transformer, evaluationMetric: String): (String, Ordering[Double]) = {
+    val modelType = getTransformerModelType(model)
+    getMetricWithOperator(modelType, evaluationMetric)
+  }
+
+  def getMetricWithOperator(model: Estimator[_], evaluationMetric: String): (String, Ordering[Double]) = {
+    val modelType = getEstimatorModelType(model)
+    getMetricWithOperator(modelType, evaluationMetric)
+  }
+
+  def getMetricWithOperator(modelType: String, evaluationMetric: String): (String, Ordering[Double]) = {
+    val chooseHighest = Ordering.Double
+    val chooseLowest = Ordering.Double.reverse
+    val (evaluationMetricColumnName, operator): (String, Ordering[Double]) = modelType match {
+      case SchemaConstants.RegressionKind => evaluationMetric match {
+        case ComputeModelStatistics.MseSparkMetric  => (ComputeModelStatistics.MseColumnName,  chooseLowest)
+        case ComputeModelStatistics.RmseSparkMetric => (ComputeModelStatistics.RmseColumnName, chooseLowest)
+        case ComputeModelStatistics.R2SparkMetric   => (ComputeModelStatistics.R2ColumnName,   chooseHighest)
+        case ComputeModelStatistics.MaeSparkMetric  => (ComputeModelStatistics.MaeColumnName,  chooseLowest)
+        case _ => throw new Exception("Metric is not supported for regressors")
+      }
+      case SchemaConstants.ClassificationKind => evaluationMetric match {
+        case ComputeModelStatistics.AucSparkMetric       => (ComputeModelStatistics.AucColumnName, chooseHighest)
+        case ComputeModelStatistics.PrecisionSparkMetric => (ComputeModelStatistics.PrecisionColumnName, chooseHighest)
+        case ComputeModelStatistics.RecallSparkMetric    => (ComputeModelStatistics.RecallColumnName, chooseHighest)
+        case ComputeModelStatistics.AccuracySparkMetric  => (ComputeModelStatistics.AccuracyColumnName, chooseHighest)
+        case _ => throw new Exception("Metric is not supported for classifiers")
+      }
+      case _ => throw new Exception("Model type not supported for evaluation")
+    }
+    (evaluationMetricColumnName, operator)
+  }
+
+  def getModelParams(model: Transformer): ParamMap = {
+    model match {
+      case reg: TrainedRegressorModel => reg.getParamMap
+      case cls: TrainedClassifierModel => cls.getParamMap
+      case evm: BestModel => getModelParams(evm.getBestModel)
+      case _ => throw new Exception("Model type not supported for evaluation")
```

Comment:
Can we try to generalize this to all sparkml estimators using the existing traits

Reply:
done!